diff --git a/extra/motioneye.init-debian b/extra/motioneye.init-debian
index 6fb627c..37b63d3 100755
--- a/extra/motioneye.init-debian
+++ b/extra/motioneye.init-debian
@@ -14,51 +14,49 @@
 
 NAME="motioneye"
 PATH="/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin"
-DAEMON="/usr/local/bin/meyectl"
+DAEMON="meyectl"
 PIDFILE="/var/run/$NAME.pid"
 DESC="motionEye server"
 USER="root"
 OPTIONS="startserver -c /etc/motioneye/motioneye.conf -l -b"
 
-. /lib/lsb/init-functions
-
-test -x $DAEMON || exit 0
+. /etc/init.d/functions
 
 RET=0
 
 case "$1" in
     start)
-        log_daemon_msg "Starting $DESC"
+        echo -n "Starting $DESC"
         if start-stop-daemon --start --oknodo --exec $DAEMON --chuid $USER -- $OPTIONS; then
-            log_end_msg 0
+            echo 0
         else
-            log_end_msg 1
+            echo 1
             RET=1
         fi
         ;;
 
     stop)
-        log_daemon_msg "Stopping $DESC"
+        echo -n "Stopping $DESC"
         if start-stop-daemon --stop --oknodo --pidfile $PIDFILE --retry 5; then
-            log_end_msg 0
+            echo 0
         else
-            log_end_msg 1
+            echo 1
             RET=1
         fi
         ;;
 
     restart|force-reload)
-        log_action_begin_msg "Restarting $DESC"
+        echo -n "Restarting $DESC"
         if $0 stop && $0 start; then
-            log_action_end_msg 0
+            echo 0
         else
-            log_action_cont_msg "(failed)"
+            echo "(failed)"
             RET=1
         fi
         ;;
 
     status)
-        status_of_proc $DAEMON $NAME
+        status $NAME
         ;;
 
     *)
