diff --git a/motioneye/handlers.py b/motioneye/handlers.py
index 31112b4..2eb974e 100644
--- a/motioneye/handlers.py
+++ b/motioneye/handlers.py
@@ -26,7 +26,8 @@ import socket
 import subprocess

 from tornado.ioloop import IOLoop
-from tornado.web import RequestHandler, StaticFileHandler, HTTPError, asynchronous
+from tornado.web import RequestHandler, StaticFileHandler, HTTPError
+from tornado.gen import coroutine

 from motioneye import config
 from motioneye import mediafiles
@@ -261,7 +262,7 @@ class ManifestHandler(BaseHandler):


 class ConfigHandler(BaseHandler):
-    @asynchronous
+    @coroutine
     def get(self, camera_id=None, op=None):
         config.invalidate_monitor_commands()

@@ -283,7 +284,7 @@ class ConfigHandler(BaseHandler):
         else:
             raise HTTPError(400, 'unknown operation')

-    @asynchronous
+    @coroutine
     def post(self, camera_id=None, op=None):
         if camera_id is not None:
             camera_id = int(camera_id)
@@ -908,7 +909,7 @@ class PictureHandler(BaseHandler):
     def compute_etag(self):
         return None

-    @asynchronous
+    @coroutine
     def get(self, camera_id, op, filename=None, group=None):
         if camera_id is not None:
             camera_id = int(camera_id)
@@ -939,7 +940,7 @@ class PictureHandler(BaseHandler):
         else:
             raise HTTPError(400, 'unknown operation')

-    @asynchronous
+    @coroutine
     def post(self, camera_id, op, filename=None, group=None):
         if group == '/':  # ungrouped
             group = ''
@@ -1393,7 +1394,7 @@ class PictureHandler(BaseHandler):


 class MovieHandler(BaseHandler):
-    @asynchronous
+    @coroutine
     def get(self, camera_id, op, filename=None):
         if camera_id is not None:
             camera_id = int(camera_id)
@@ -1409,7 +1410,7 @@ class MovieHandler(BaseHandler):
         else:
             raise HTTPError(400, 'unknown operation')

-    @asynchronous
+    @coroutine
     def post(self, camera_id, op, filename=None, group=None):
         if group == '/':  # ungrouped
             group = ''
@@ -1561,7 +1562,7 @@ class MoviePlaybackHandler(StaticFileHandler, BaseHandler):
     if not os.path.exists(tmpdir):
         os.mkdir(tmpdir)

-    @asynchronous
+    @coroutine
     @BaseHandler.auth()
     def get(self,  camera_id, filename=None, include_body=True):
         logging.debug('downloading movie %(filename)s of camera %(id)s' % {
@@ -1640,7 +1641,7 @@ class MovieDownloadHandler(MoviePlaybackHandler):


 class ActionHandler(BaseHandler):
-    @asynchronous
+    @coroutine
     def post(self, camera_id, action):
         camera_id = int(camera_id)
         if camera_id not in config.get_camera_ids():
