diff --git a/motioneye/v4l2ctl.py b/motioneye/v4l2ctl.py
index 79448ca..4d4e33a 100644
--- a/motioneye/v4l2ctl.py
+++ b/motioneye/v4l2ctl.py
@@ -51,7 +51,7 @@ def list_devices():
     try:
         output = ''
         started = time.time()
-        p = subprocess.Popen(['v4l2-ctl', '--list-devices'], stdout=subprocess.PIPE, bufsize=1)
+        p = subprocess.Popen(['v4l2-ctl', '--list-devices'], stdout=subprocess.PIPE, bufsize=0)
 
         fd = p.stdout.fileno()
         fl = fcntl.fcntl(fd, fcntl.F_GETFL)
@@ -59,15 +59,15 @@ def list_devices():
 
         while True:
             try:
-                data = p.stdout.read(1024)
+                data = p.stdout.readline()
                 if not data:
                     break
 
             except IOError:
-                data = ''
+                data = b''
                 time.sleep(0.01)
 
-            output += data
+            output += data.decode('utf-8')
 
             if len(output) > 10240:
                 logging.warn('v4l2-ctl command returned more than 10k of output')
@@ -128,7 +128,7 @@ def list_resolutions(device):
     cmd = 'v4l2-ctl -d %(device)s --list-formats-ext | grep -vi stepwise | grep -oE "[0-9]+x[0-9]+" || true' % {
             'device': pipes.quote(device)}
 
-    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, bufsize=1)
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, bufsize=0)
 
     fd = p.stdout.fileno()
     fl = fcntl.fcntl(fd, fcntl.F_GETFL)
@@ -136,15 +136,15 @@ def list_resolutions(device):
 
     while True:
         try:
-            data = p.stdout.read(1024)
+            data = p.stdout.readline()
             if not data:
                 break
 
         except IOError:
-            data = ''
+            data = b''
             time.sleep(0.01)
 
-        output += data
+        output += data.decode('utf-8')
 
         if len(output) > 10240:
             logging.warn('v4l2-ctl command returned more than 10k of output')
