diff --git a/Makefile.am b/Makefile.am
index d50bcd5..b072f72 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -45,6 +45,7 @@ all-local:
 ## Update contents of sample files with the correct directory
 ###################################################################
 install-data-local:
+	mkdir -p -m666 ${DESTDIR}${sysconfdir}
 	@sed -e 's|$${prefix}|$(prefix)|' ${DESTDIR}${sysconfdir}/motionplus-dist.conf > ${DESTDIR}${sysconfdir}/motionplus-dist.conf.tmp && mv -f ${DESTDIR}${sysconfdir}/motionplus-dist.conf.tmp ${DESTDIR}${sysconfdir}/motionplus-dist.conf
 	@sed -e 's|$${prefix}|$(prefix)|' ${DESTDIR}${sysconfdir}/camera1-dist.conf > ${DESTDIR}${sysconfdir}/camera1-dist.conf.tmp && mv -f ${DESTDIR}${sysconfdir}/camera1-dist.conf.tmp ${DESTDIR}${sysconfdir}/camera1-dist.conf
 	@sed -e 's|$${prefix}|$(prefix)|' ${DESTDIR}${sysconfdir}/camera2-dist.conf > ${DESTDIR}${sysconfdir}/camera2-dist.conf.tmp && mv -f ${DESTDIR}${sysconfdir}/camera2-dist.conf.tmp ${DESTDIR}${sysconfdir}/camera2-dist.conf
